From 6400323373c08015274108e4b8c30e7f3418e6c9 Mon Sep 17 00:00:00 2001
From: Hyunsu Cho <phcho@nvidia.com>
Date: Tue, 12 Aug 2025 12:07:36 -0700
Subject: [PATCH] Patch for CUDA 13

---
 cpp/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index 2fc9237..a594064 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -15,8 +15,8 @@
 #=============================================================================

 cmake_minimum_required(VERSION 3.21 FATAL_ERROR)
-set(rapids-cmake-version 25.06)
-file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-25.06/RAPIDS.cmake
+set(rapids-cmake-version 25.10)
+file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-25.10/RAPIDS.cmake
     ${CMAKE_BINARY_DIR}/RAPIDS.cmake)
 include(${CMAKE_BINARY_DIR}/RAPIDS.cmake)
 include(rapids-cmake)
--
2.43.0

